jpo-ode-plugins/src/test/java/us/dot/its/jpo/ode/plugin/j2735/builders/TimBuilderTest.java

```diff
@@ -0,0 +1,30 @@
+		assertEquals(expected, actualTimString); 
+	}
+}
```

Comment:
suggestion(non-blocking): when testing we don't need to explicitly catch and fail on exceptions. We can just let JUnit handle that by adding a `throws` to the test method signature

```suggestion
class TimBuilderTest {
  @Test
	void shouldTranslateTim() throws XmlUtilsException {

		JsonNode jsonMap = XmlUtils.toObjectNode(
					"<OdeAsn1Data><metadata><logFileName/><recordType>timMsg</recordType><securityResultCode>success</securityResultCode><receivedMessageDetails/><encodings><encodings><elementName>unsecuredData</elementName><elementType>MessageFrame</elementType><encodingRule>UPER</encodingRule></encodings></encodings><payloadType>us.dot.its.jpo.ode.model.OdeAsn1Payload</payloadType><serialId><streamId>4d476fdf-9103-464d-88c5-5de24b9f3ef9</streamId><bundleSize>1</bundleSize><bundleId>0</bundleId><recordId>0</recordId><serialNumber>0</serialNumber></serialId><odeReceivedAt>2024-11-21T01:17:47.998Z</odeReceivedAt><schemaVersion>7</schemaVersion><maxDurationTime>0</maxDurationTime><recordGeneratedAt/><recordGeneratedBy>RSU</recordGeneratedBy><sanitized>false</sanitized><odePacketID/><odeTimStartDateTime/><asn1>001F810A701607988D44FF260FFFD00C010F775D9B0B01C2715BDE59674A61AFFFF93F4310B021C0A007F91A54ED2088C10346A820F2C7A105164CE000000002715BDE59674A61A9388CE00021100CFA9460EE711C707B8C97B45536D12F62A75A8B8E0F329509879F9AF683B375AAC1B8AEAF70D60B3DA000C80218312D11081EEEBB36301C2715DFCC1673BB4FFFFF93F4310B021C0A007F99A54ED2088C10346A820E8CB8722083CB1E8CBC905164CE000000002715DFCC1673BB4F9388C7C0021160A5A4000E1D8955078014AA83CFCA84C1F47D5ED1521A3AF68AAD25DED055B4CF7B42AF9D7D5515A5DBFB40AE11A05EC559B908E42A8888555001A00430601C312D11080D404F775D9B00358C2614C511ABFC4ED8D5830858CC42A5B1B0023B5A84F612289408C096D7DFC084AA26FA6023C7D787B119AC3920090449C099FE4A4DF873D6B46EC2FC98916754931A4C5A500000000134E749ACB3149E209C46003C10B02211F450076888E5258B044869CC2E211098FF7108FC9E41897CCDC1704B9FCE4D42435E7023127413C78893C4DF7E848769083412036397111DCCC1B089CBA66C4435093DE2205BCA4D7089754160D73EF8A0E8441EB237825E3184501296941CB88E972102448D87064A3520F20A170A0350E408AABE263849C0B0D302157C8CC612CE3C3CA893FB247C420300</asn1><originIp>172.18.0.1</originIp></metadata><payload><dataType>MessageFrame</dataType><data><MessageFrame><messageId>31</messageId><value><TravelerInformation><msgCnt>1</msgCnt><timeStamp>395160</timeStamp><packetID>8D44FF260FFFD00C01</packetID><urlB>null</urlB><dataFrames><TravelerDataFrame><notUsed>0</notUsed><frameType><advisory/></frameType><msgId><roadSignID><position><lat>411472587</lat><long>-1046513098</long></position><viewAngle>1111111111111111</viewAngle><mutcdCode><warning/></mutcdCode></roadSignID></msgId><startYear>2024</startYear><startTime>401760</startTime><durationTime>8640</durationTime><priority>5</priority><notUsed1>0</notUsed1><regions><GeographicalPath><name>RSZ D0 45 Arch Q23</name><id><region>0</region><id>0</id></id><anchor><lat>411472587</lat><long>-1046513098</long></anchor><laneWidth>5000</laneWidth><directionality><both/></directionality><closedPath><true/></closedPath><direction>0111000000000000</direction><description><path><scale>0</scale><offset><ll><nodes><NodeLL><delta><node-LL1><lon>1274</lon><lat>326</lat></node-LL1></delta></NodeLL><NodeLL><delta><node-LL2><lon>5944</lon><lat>910</lat></node-LL2></delta></NodeLL><NodeLL><delta><node-LL2><lon>7052</lon><lat>1517</lat></node-LL2></delta></NodeLL><NodeLL><delta><node-LL3><lon>10678</lon><lat>2427</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>10074</lon><lat>2958</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL2><lon>6548</lon><lat>2579</lat></node-LL2></delta></NodeLL><NodeLL><delta><node-LL2><lon>6649</lon><lat>3034</lat></node-LL2></delta></NodeLL><NodeLL><delta><node-LL2><lon>5742</lon><lat>3414</lat></node-LL2></delta></NodeLL><NodeLL><delta><node-LL2><lon>3627</lon><lat>2807</lat></node-LL2></delta></NodeLL><NodeLL><delta><node-LL2><lon>2821</lon><lat>1972</lat></node-LL2></delta></NodeLL></nodes></ll></offset></path></description></GeographicalPath></regions><notUsed2>0</notUsed2><notUsed3>0</notUsed3><content><speedLimit><SEQUENCE><item><itis>268</itis></item></SEQUENCE><SEQUENCE><item><itis>12589</itis></item></SEQUENCE><SEQUENCE><item><itis>8720</itis></item></SEQUENCE></speedLimit></content><url>null</url></TravelerDataFrame><TravelerDataFrame><notUsed>0</notUsed><frameType><advisory/></frameType><msgId><roadSignID><position><lat>411489944</lat><long>-1046633312</long></position><viewAngle>1111111111111111</viewAngle><mutcdCode><warning/></mutcdCode></roadSignID></msgId><startYear>2024</startYear><startTime>401760</startTime><durationTime>8640</durationTime><priority>5</priority><notUsed1>0</notUsed1><regions><GeographicalPath><name>RSZ D0 45 Ahead Archer Q23</name><id><region>0</region><id>0</id></id><anchor><lat>411489944</lat><long>-1046633312</long></anchor><laneWidth>5000</laneWidth><directionality><both/></directionality><closedPath><false/></closedPath><direction>0011111000000000</direction><description><path><scale>0</scale><offset><ll><nodes><NodeLL><delta><node-LL1><lon>602</lon><lat>-1024</lat></node-LL1></delta></NodeLL><NodeLL><delta><node-LL2><lon>4332</lon><lat>-3414</lat></node-LL2></delta></NodeLL><NodeLL><delta><node-LL2><lon>6145</lon><lat>-3414</lat></node-LL2></delta></NodeLL><NodeLL><delta><node-LL2><lon>6649</lon><lat>-3034</lat></node-LL2></delta></NodeLL><NodeLL><delta><node-LL2><lon>7455</lon><lat>-2579</lat></node-LL2></delta></NodeLL><NodeLL><delta><node-LL3><lon>9268</lon><lat>-2579</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>11081</lon><lat>-2124</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>11686</lon><lat>-1062</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>12189</lon><lat>-683</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>13499</lon><lat>-152</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>14406</lon><lat>379</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>11484</lon><lat>1138</lat></node-LL3></delta></NodeLL><NodeLL><delta><node-LL3><lon>10376</lon><lat>1365</lat></node-LL3></delta></NodeLL></nodes></ll></offset></path></description></GeographicalPath></regions><notUsed2>0</notUsed2><notUsed3>0</notUsed3><content><speedLimit><SEQUENCE><item><itis>268</itis></item></SEQUENCE><SEQUENCE><item><itis>12302</itis></item></SEQUENCE><SEQUENCE><item><itis>12589</itis></item></SEQUENCE><SEQUENCE><item><itis>8720</itis></item></SEQUENCE><SEQUENCE><item><itis>13569</itis></item></SEQUENCE></speedLimit></content><url>null</url></TravelerDataFrame></dataFrames></TravelerInformation></value></MessageFrame></data></payload></OdeAsn1Data>");
		J2735Tim actualTim = TIMBuilder.genericTim(jsonMap.findValue("TravelerInformation"));	
		String actualTimString = actualTim.toString();
		String expected ="{\"msgCnt\":\"1\",\"timeStamp\":\"395160\",\"packetID\":\"8D44FF260FFFD00C01\",\"dataFrames\":[{\"notUsed\":\"0\",\"frameType\":\"advisory\",\"msgId\":{\"roadSignID\":{\"position\":{\"latitude\":41.1472587,\"longitude\":-104.6513098},\"viewAngle\":\"1111111111111111\",\"mutcdCode\":\"warning\"}},\"startYear\":\"2024\",\"startTime\":\"401760\",\"durationTime\":\"8640\",\"priority\":\"5\",\"notUsed1\":\"0\",\"regions\":[{\"name\":\"RSZ D0 45 Arch Q23\",\"id\":{\"region\":0,\"id\":0},\"anchor\":{\"latitude\":41.1472587,\"longitude\":-104.6513098},\"laneWidth\":5000,\"directionality\":\"both\",\"closedPath\":true,\"description\":{\"path\":{\"scale\":0,\"offset\":{\"ll\":{\"nodes\":[{\"delta\":{\"nodeLL1\":{\"lon\":1274,\"lat\":326}}},{\"delta\":{\"nodeLL2\":{\"lon\":5944,\"lat\":910}}},{\"delta\":{\"nodeLL2\":{\"lon\":7052,\"lat\":1517}}},{\"delta\":{\"nodeLL3\":{\"lon\":10678,\"lat\":2427}}},{\"delta\":{\"nodeLL3\":{\"lon\":10074,\"lat\":2958}}},{\"delta\":{\"nodeLL2\":{\"lon\":6548,\"lat\":2579}}},{\"delta\":{\"nodeLL2\":{\"lon\":6649,\"lat\":3034}}},{\"delta\":{\"nodeLL2\":{\"lon\":5742,\"lat\":3414}}},{\"delta\":{\"nodeLL2\":{\"lon\":3627,\"lat\":2807}}},{\"delta\":{\"nodeLL2\":{\"lon\":2821,\"lat\":1972}}}]}}}}}],\"notUsed2\":\"0\",\"notUsed3\":\"0\",\"content\":{\"speedLimit\":{\"SEQUENCE\":[{\"item\":{\"itis\":\"268\"}},{\"item\":{\"itis\":\"12589\"}},{\"item\":{\"itis\":\"8720\"}}]}}},{\"notUsed\":\"0\",\"frameType\":\"advisory\",\"msgId\":{\"roadSignID\":{\"position\":{\"latitude\":41.1489944,\"longitude\":-104.6633312},\"viewAngle\":\"1111111111111111\",\"mutcdCode\":\"warning\"}},\"startYear\":\"2024\",\"startTime\":\"401760\",\"durationTime\":\"8640\",\"priority\":\"5\",\"notUsed1\":\"0\",\"regions\":[{\"name\":\"RSZ D0 45 Ahead Archer Q23\",\"id\":{\"region\":0,\"id\":0},\"anchor\":{\"latitude\":41.1489944,\"longitude\":-104.6633312},\"laneWidth\":5000,\"directionality\":\"both\",\"closedPath\":false,\"description\":{\"path\":{\"scale\":0,\"offset\":{\"ll\":{\"nodes\":[{\"delta\":{\"nodeLL1\":{\"lon\":602,\"lat\":-1024}}},{\"delta\":{\"nodeLL2\":{\"lon\":4332,\"lat\":-3414}}},{\"delta\":{\"nodeLL2\":{\"lon\":6145,\"lat\":-3414}}},{\"delta\":{\"nodeLL2\":{\"lon\":6649,\"lat\":-3034}}},{\"delta\":{\"nodeLL2\":{\"lon\":7455,\"lat\":-2579}}},{\"delta\":{\"nodeLL3\":{\"lon\":9268,\"lat\":-2579}}},{\"delta\":{\"nodeLL3\":{\"lon\":11081,\"lat\":-2124}}},{\"delta\":{\"nodeLL3\":{\"lon\":11686,\"lat\":-1062}}},{\"delta\":{\"nodeLL3\":{\"lon\":12189,\"lat\":-683}}},{\"delta\":{\"nodeLL3\":{\"lon\":13499,\"lat\":-152}}},{\"delta\":{\"nodeLL3\":{\"lon\":14406,\"lat\":379}}},{\"delta\":{\"nodeLL3\":{\"lon\":11484,\"lat\":1138}}},{\"delta\":{\"nodeLL3\":{\"lon\":10376,\"lat\":1365}}}]}}}}}],\"notUsed2\":\"0\",\"notUsed3\":\"0\",\"content\":{\"speedLimit\":{\"SEQUENCE\":[{\"item\":{\"itis\":\"268\"}},{\"item\":{\"itis\":\"12302\"}},{\"item\":{\"itis\":\"12589\"}},{\"item\":{\"itis\":\"8720\"}},{\"item\":{\"itis\":\"13569\"}}]}}}]}";
		assertEquals(expected, actualTimString); 
	}
}
```